NeoML/src/Dnn/Layers/MobileNetV2BlockLayer.cpp

```diff
@@ -184,44 +185,67 @@ void CMobileNetV2BlockLayer::Reshape()
 	}
 	outputDescs[0].SetDimSize( BD_Channels, outputChannels );
 
+	if( InputsMayBeOverwritten() && inputDescs[0].HasEqualDimensions( outputDescs[0] ) ) {
+		NeoAssert( stride == 1 );
+		EnableInPlace( true );
+	}
+
+	recreateConvDesc( expandedChannels );
+	recreateRowwiseDesc( expandedChannels );
+}
+
+void CMobileNetV2BlockLayer::recreateConvDesc( int expandedChannels )
+{
 	if( convDesc != nullptr ) {
 		delete convDesc;
 		convDesc = nullptr;
 	}
+
 	CBlobDesc channelwiseInputDesc = inputDescs[0];
 	channelwiseInputDesc.SetDimSize( BD_Channels, expandedChannels );
 	CBlobDesc channelwiseOutputDesc = outputDescs[0];
 	channelwiseOutputDesc.SetDimSize( BD_Channels, expandedChannels );
-	CBlobDesc freeTermDesc = paramBlobs[P_ChannelwiseFreeTerm] != nullptr
-		? paramBlobs[P_ChannelwiseFreeTerm]->GetDesc() : CBlobDesc();
+	CBlobDesc freeTermDesc = ( paramBlobs[P_ChannelwiseFreeTerm] == nullptr ) ? CBlobDesc{}
+		: paramBlobs[P_ChannelwiseFreeTerm]->GetDesc();
+
 	convDesc = MathEngine().InitBlobChannelwiseConvolution( channelwiseInputDesc, 1, 1, stride, stride,
 		paramBlobs[P_ChannelwiseFilter]->GetDesc(),
-		paramBlobs[P_ChannelwiseFreeTerm] != nullptr ? &freeTermDesc : nullptr, channelwiseOutputDesc );
-
-	if( InputsMayBeOverwritten() && inputDescs[0].HasEqualDimensions( outputDescs[0] ) ) {
```

Comment:
Why did you disable in-place?

Reply:
It is not disabled)
These lines are moved out of new method `recreateConvDesc()` from 201-204 lines to 187-190 lines in `Reshape()`.